EspressoSystems/HotShot
# [Auto Bench] Scripts of Running Benchmarks in AWS

Closes #2723 #2354 .



### This PR: 
- automates the flow we have to do to get benchmarks in the cloud.
- includes benchmark results for init run. See `scripts/benchmarks_results/results_init_run.csv`.
- fixes `run-config.toml` and `post_pub_key` for examples after Builder and vbs get merged in.
- cleans up some files by adding to gitignore.






### This PR does not (but going to do next): 
- consider any vid-related factor.
- add runs for push-based CDN, libp2p and combined network yet (as now it only include runs for webserver).
- use nginx while this is needed for `total_nodes >= 500`.
- tune the parameters according to the results.
- finish ci integration.
- add runs for `tokio` as it only support `async_std` now.



### Key places to review: 
`scripts/aws_ecs_benchmarks.sh`



### How to test this PR:
- Log into Github on the EC2 machine.
- Get things to run `HotShot` installed (like Rust) and clone `HotShot`.
- Run `./scripts/aws_ecs_benchmarks.sh`.
- Three ways to gather the results
    - (recommended) check out`scripts/benchmarks_results/results.csv` on EC2, where you should have organized overall results, 
    - check out Datadog under `host:/hotshot/` where you only have metrics for each individual validator and it's hard to track,
    - wait for the output in terminal where the results are not that organized, also hard to track.
